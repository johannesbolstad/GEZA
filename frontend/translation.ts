import * as Speech from 'expo-speech';
import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

/**
 * A function to speak out loud the text in a given language
 *
 * @param languageToSpeak the language is the preferred language
 * @param thingToSay thingTosay is the text that will be spoken
 */
export const speakUp = (languageToSpeak: string, thingToSay: string) => {
    if (languageToSpeak === 'en') {
        Speech.speak(thingToSay, {
            language: 'en-IN'
        });
    }
    if (languageToSpeak === 'no') {
        Speech.speak(thingToSay, {
            language: 'nb-NO'
        });
    }
    if (languageToSpeak === 'fr') {
        Speech.speak(thingToSay, {
            language: 'fr-FR'
        });
    }
    if (languageToSpeak === 'es') {
        Speech.speak(thingToSay, {
            language: 'es-ES'
        });
    }
    if (languageToSpeak === 'de') {
        Speech.speak(thingToSay, {
            language: 'de-DE'
        });
    }
    if (languageToSpeak === 'عر') {
        Speech.speak(thingToSay, {
            language: 'ar-XA'
        });
    }
    if (languageToSpeak === 'ук') {
        Speech.speak(thingToSay, {
            language: 'uk-UA'
        });
    }
};

// contains all words translated into different languages.
const resources = {
    en: {
        translation: {
            //Navbar
            Home: 'Home',
            Groups: 'Groups',
            Messages: 'Messages',
            Wiki: 'Wiki',
            //Wiki categories
            'New in Norway': 'New in Norway',
            Family: 'Family',
            Health: 'Health',
            Economy: 'Economy',
            Sparetime: 'Sparetime',
            SOS: 'SOS',
            // Wiki subcategories
            Emergency: 'Emergency',
            Coronavirus: 'Coronavirus',
            Hospital: 'Hospital',
            'Useful resources': 'Useful resources',
            'Brief history': 'Brief history',
            Job: 'Job',
            Education: 'Education',
            Kindergarten: 'Kindergarten',
            School: 'School',
            // Buttons
            Continue: 'Continue',
            'register a user': 'register a user',
            'Continue without user': 'Continue without user',
            //Sign up screen
            'User Name': 'User Name',
            Location: 'Location',
            Language: 'Language',
            'Year of birth': 'Year of birth',
            'Years lived in Norway': 'Years lived in Norway',
            'Sign up': 'Sign up',
            Back: 'Back',
            Male: 'Male',
            Female: 'Female',
            Other: 'Other',

            //Make Group
            'Create new Group': 'Create new Group',
            'Choose Image': 'Choose Image',
            'Group name': 'Group name',
            Description: 'Description',
            'Select language': 'Select language',
            'Make group': 'Make group',
            Title: 'Title',

            'SHOW MORE': 'SHOW MORE',
            'SHOW LESS': 'SHOW LESS',
            'Do you want to join': 'Do you want to join',
            Cancel: 'Cancel',
            Join: 'Join',
            'Create new post': 'Create new post',
            'Emergency numbers': 'Emergency numbers',
            Comment: 'Comment',
            'Create comment': 'Create comment',
            'Write comment': 'Write comment',
            'Type a message': 'Type a message',
            'New message': 'New message',
            Search: 'Search',
            'Save changes': 'Save changes',
            'Read more': 'Read more',
            'Read less': 'Read less',
            'Write some description': 'Write some description',
            'Enter GroupPost title': 'Enter GroupPost title',
            'Create post': 'Create post',
            'Name of recipient': 'Name of recipient'
        }
    },
    ук: {
        translation: {
            //Navbar
            Home: 'Додому',
            Groups: 'Групи',
            Messages: 'Повідомлення',
            Wiki: 'Вікі',
            //Wiki categories
            'New in Norway': 'Нові в Норвегії',
            Family: "Сім'я",
            Health: "Здоров'я",
            Economy: 'Економіка',
            Sparetime: 'Вільний час',
            SOS: 'сигнал про небезпеку',
            // Wiki subcategories
            Emergency: 'Надзвичайна ситуація',
            Coronavirus: 'Коронавірус',
            Hospital: 'Лікарня',
            'Useful resources': 'Корисні ресурси',
            'Brief history': 'Коротка історія',
            Job: 'Робота',
            Education: 'Освіта',
            Kindergarten: 'дитячий садок',
            School: 'Школа',
            // Buttons
            Continue: 'Продовжуйте',
            'register a user': 'Зареєструвати користувача',
            'Continue without user': 'Продовжити без користувача',
            //Sign up screen
            'User Name': "Ім'я користувача",
            Location: 'Розташування',
            'Year of birth': 'Рік народження',
            'Years lived in Norway': 'Роки проживання у Норвегії',
            'Sign up': 'підпиши',
            Back: 'Назад',
            Male: 'Чоловічий',
            Female: 'Жіночий',
            Other: 'Інший',

            //Make Group
            'Create new Group': 'Створити нову групу',
            'Choose Image': 'Виберіть Зображення',
            'Group name': 'Назва групи',
            Description: 'Опис',
            'Select language': 'Оберіть мову',
            'Make group': 'Створити групу',
            Title: 'титул',
            'SHOW MORE': 'ПОКАЗАТИ БІЛЬШЕ',
            'SHOW LESS': 'ПОКАЗАТИ МЕНШЕ',
            'Do you want to join': 'Хочете приєднатися',
            Cancel: 'Скасувати',
            Join: 'Приєднуйтесь',
            'Create new post': 'Створити новий пост',
            'Emergency numbers': 'Номери екстреної допомоги',
            Comment: 'коментар',
            'Create comment': 'Створити коментар',
            'Write comment': 'Напишіть коментар',
            'Type a message': 'Введіть повідомлення',
            'New message': 'Нове повідомлення',
            Search: 'Пошук',
            'Save changes': 'Зберегти зміни',
            'Read more': 'Детальніше',
            'Read less': 'Читайте менше',
            'Write some description': 'Напишіть якийсь опис',
            'Enter GroupPost title': 'Введіть назву допису групи',
            'Create post': 'Створити допис',
            'Name of recipient': "Ім'я одержувача"
        }
    },
    no: {
        translation: {
            Home: 'Hjem',
            Groups: 'Grupper',
            Messages: 'Meldinger',
            Wiki: 'Wiki',
            'New in Norway': 'Ny i Norge',
            Family: 'Familie',
            Health: 'Helse',
            Economy: 'Økonomi',
            Sparetime: 'Fritid',
            // Wiki subcategories
            Emergency: 'Nødsituasjon',
            Coronavirus: 'Coronavirus',
            Hospital: 'Sykehus',
            'Useful resources': 'Nyttige ressurser',
            'Brief history': 'Kort historie',
            Job: 'Jobb',
            Education: 'Utdanning',
            Kindergarten: 'Barnehage',
            School: 'Skole',
            Continue: 'Fortsett',
            'register a user': 'Registrer en bruker',
            'Continue without user': 'fortsett uten bruker',
            'User Name': 'Brukernavn',
            Location: 'Sted',
            Language: 'Språk',
            'Year of birth': 'Fødselsår',
            'Years lived in Norway': 'Antall år i Norge',
            'Sign up': 'Registrer',
            Back: 'Tilbake',
            Male: 'Mann',
            Female: 'Kvinne',
            Other: 'Andre',
            'Create new Group': 'Opprett ny gruppe',
            'Choose Image': 'Velg bilde',
            'Group name': 'Gruppenavn',
            Description: 'Beskrivelse',
            'Select language': 'Velg språk',
            Title: 'Tittel',
            'Make group': 'Lag gruppe',
            'SHOW MORE': 'VIS MER',
            'SHOW LESS': 'VIS MINDRE',
            'Do you want to join': 'Oensker du aa bli med i',
            Cancel: 'Avbryt',
            Join: 'Bli med',
            'Create new post': 'Opprett nytt innlegg',
            'Emergency numbers': 'Nødnumre',
            Comment: 'Kommentere',
            'Create comment': 'Legg til kommentar',
            'Write comment': 'Skriv en kommentar',
            'Type a message': 'Skriv inn en melding',
            'New message': 'Ny melding',
            Search: 'Søk',
            'Save changes': 'Lagre endringer',
            'Read more': 'Les mer',
            'Read less': 'Les mindre',
            'Write some description': 'Skriv en beskrivelse',
            'Enter GroupPost title': 'Skriv inn Group Post-tittel',
            'Create post': 'Opprett innlegg',
            'Name of recipient': 'Navn på mottaker'
        }
    },
    fr: {
        translation: {
            Home: 'Maison',
            Groups: 'Groupes',
            Messages: 'Messages',
            Wiki: 'Wiki',
            'New in Norway': 'Nouveau en Norvège',
            Family: 'Famille',
            Health: 'Santé',
            Economy: 'Économie',
            Sparetime: 'Temps libre',
            // Wiki subcategories
            Emergency: 'Urgence',
            Coronavirus: 'Coronavirus',
            Hospital: 'Hôpital',
            'Useful resources': 'Ressources utiles',
            'Brief history': 'Bref historique',
            Job: 'Emploi',
            Education: 'Éducation',
            Kindergarten: "Jardin d'enfants",
            School: "L'école",
            Continue: 'Continuez',
            'register a user': 'Enregistrer un utilisateur',
            'Continue without user': 'Continuer sans utilisateur',
            'User Name': "Nom d'utilisateur",
            Location: 'Lieu',
            Language: 'Langue',
            'Year of birth': 'Année de naissance',
            'Years lived in Norway': 'Des années en Norvège',
            'Sign up': "S'inscrire",
            Back: 'Arrière',
            Male: 'Mâle',
            Female: 'Femelle',
            Other: 'Autre',
            'Create new Group': 'Créer un nouveau groupe',
            'Choose Image': 'Choisir une image',
            'Group name': 'Nom de groupe',
            Description: 'La description',
            'Select language': 'Choisir la langue',
            'Make group': 'Créer un groupe',
            Title: 'Titre',
            'SHOW MORE': 'MONTRE PLUS',
            'SHOW LESS': 'MONTRER MOINS',
            'Do you want to join': 'Veux tu rejoindre',
            Cancel: 'Annuler',
            Join: 'Rejoindre',
            'Create new post': 'Créer un nouveau message',
            'Emergency numbers': "Numéros d'urgence",
            Comment: 'Commenter',
            'Create comment': 'Créer un commentaire',
            'Write comment': 'Rédiger un commentaire',
            'Type a message': 'Tapez un message',
            'New message': 'Nouveau message',
            Search: 'Recherche',
            'Save changes': 'Sauvegarder les modifications',
            'Read more': 'Lire la suite',
            'Read less': 'Lire moins',
            'Write some description': 'Rédigez une description',
            'Enter GroupPost title': 'Entrez le titre de GroupPost',
            'Create post': 'Créer un message',
            'Name of recipient': 'Nom du récipient'
        }
    },

    es: {
        translation: {
            Home: 'Casa',
            Groups: 'Grupos',
            Messages: 'Mensajes',
            Wiki: 'Wiki',
            'New in Norway': 'Nuevo en Noruega',
            Family: 'Familia',
            Health: 'Salud',
            Economy: 'Economía',
            Sparetime: 'Tiempo libre',
            // Wiki subcategories
            Emergency: 'Emergencia',
            Coronavirus: 'Coronavirus',
            Hospital: 'Hospital',
            'Useful resources': 'Recursos útiles',
            'Brief history': 'Breve historia',
            Job: 'Trabajo',
            Education: 'Educación',
            Kindergarten: 'Jardín de infancia',
            School: 'Colegio',
            Continue: 'Continuar',
            'register a user': 'Registrar un usuario',
            'Continue without user': 'Registrar sin usuario',
            'User Name': 'Nombre de usuario',
            Location: 'La localización',
            Language: 'Idioma',
            'Year of birth': 'Año de nacimiento',
            'Years lived in Norway': 'Años vividos en Noruega',
            'Sign up': 'Inscribirse',
            Back: 'Atrás',
            Male: 'Masculino',
            Female: 'Masculina',
            Other: 'Otro',
            'Create new Group': 'Crear nuevo grupo',
            'Choose Image': 'elegir imagen',
            'Group name': 'Nombre del grupo',
            Description: 'Descripción',
            'Select language': 'Elegir el idioma',
            'Make group': 'crear grupo',
            Title: 'Título',
            'SHOW MORE': 'MOSTRAR MÁS',
            'SHOW LESS': 'MUESTRA MENOS',
            'Do you want to join': 'Quieres unirte',
            Cancel: 'Cancelar',
            Join: 'Entrar',
            'Create new post': 'Crear nueva publicación',
            'Emergency numbers': 'Números de emergencia',
            Comment: 'Comentario',
            'Create comment': 'Crear comentario',
            'Write comment': 'escribir comentario',
            'Type a message': 'Escriba un mensaje',
            'New message': 'Nuevo mensaje',
            Search: 'Búsqueda',
            'Save changes': 'Guardar cambios',
            'Read more': 'Lee mas',
            'Read less': 'Leer menos',
            'Write some description': 'Escribe alguna descripción',
            'Enter GroupPost title':
                'Ingrese el título de la publicación del grupo',
            'Create post': 'Crear publicación',
            'Name of recipient': 'Nombre del destinatario'
        }
    },
    عر: {
        translation: {
            Home: 'مسكن',
            Groups: 'مجموعات',
            Messages: 'رسائل',
            Wiki: 'ويكي',
            //Wiki categories
            'New in Norway': 'جديد في النرويج',
            Family: 'الأسرة',
            Health: 'الصحة',
            Economy: 'اقتصاد',
            Sparetime: 'وقت الفراغ',
            SOS: 'نِدَاءُ اسْتِغَاثَة',
            // Wiki subcategories
            Emergency: 'طارئ',
            Coronavirus: 'فيروس كورونا',
            Hospital: 'مستشفى',
            'Useful resources': 'موارد مفيدة',
            'Brief history': 'نبذة تاريخية',
            Job: 'وظيفة',
            Education: 'تعليم',
            Kindergarten: 'روضة أطفال',
            School: 'مدرسة',
            Continue: 'يكمل',
            'register a user': 'تسجيل مستخدم',
            'Continue without user': 'تواصل بدون مستخدم',
            'User Name': 'اسم المستخدم',
            Location: 'موقع',
            Language: 'لغة',
            'Year of birth': 'سنة الميلاد',
            'Years lived in Norway': 'عاش سنوات في النرويج',
            'Sign up': 'اشتراك',
            Back: 'خلف',
            Male: 'ذكر',
            Female: 'أنثى',
            Other: 'آخر',
            'Create new Group': 'إنشاء مجموعة جديدة',
            'Choose Image': 'اختر صورة',
            'Group name': 'أسم المجموعة',
            Description: 'وصف',
            'Select language': 'اختار اللغة',
            'Make group': 'اصنع مجموعة',
            Title: 'عنوان',
            'SHOW MORE': 'أظهر المزيد',
            'SHOW LESS': 'تظهر أقل',
            'Do you want to join': 'هل تود الانضمام',
            Cancel: 'يلغي',
            Join: 'ينضم',
            'Create new post': 'إنشاء منشور جديد',
            'Emergency numbers': 'أرقام الطوارئ',
            Comment: 'تعليق',
            'Create comment': 'إنشاء تعليق',
            'Write comment': 'اكتب تعليق',
            'Type a message': 'اكتب رسالة',
            'New message': 'رسالة جديدة',
            Search: 'بحث',
            'Save changes': 'احفظ التغييرات',
            'Read more': 'اقرأ أكثر',
            'Read less': 'أقرأ أقل',
            'Create post': 'إنشاء وظيفة',
            'Enter GroupPost title': 'أدخل عنوان GroupPost',
            'Write some description': 'اكتب بعض الوصف',
            'Name of recipient': 'اسم المستلم'
        }
    },
    de: {
        translation: {
            Home: 'Zuhause',
            Groups: 'Gruppen',
            Messages: 'Nachrichten',
            Wiki: 'Wiki',
            'New in Norway': 'Neu in Norwegen',
            Family: 'Familie',
            Health: 'Die Gesundheit',
            Economy: 'Wirtschaft',
            Sparetime: 'Freizeit',
            // Wiki subcategories
            Emergency: 'Notfall',
            Coronavirus: 'Coronavirus',
            Hospital: 'Krankenhaus',
            'Useful resources': 'Nützliche Ressourcen',
            'Brief history': 'Kurze Geschichte',
            Job: 'Arbeit',
            Education: 'Bildung',
            Kindergarten: 'Kindergarten',
            School: 'Schule',
            Continue: 'Fortsetzen',
            'register a user': 'Registrieren Sie einen Benutzer',
            'Continue without user': 'Ohne Benutzer fortfahren',
            'User Name': 'Nutzername',
            Location: 'Standort',
            Language: 'Sprache',
            'Year of birth': 'Geburtsjahr',
            'Years lived in Norway': 'Jahre in Norwegen gelebt',
            'Sign up': 'Anmelden',
            Back: 'Zurück',
            Male: 'Männlich',
            Female: 'Weiblich',
            Other: 'Andere',
            'Create new Group': 'Neue Gruppe erstellen',
            'Choose Image': 'Wählen Sie Bild',
            'Group name': 'Gruppenname',
            Description: 'Beschreibung',
            'Select language': 'Select language',
            'Make group': 'Gruppe machen',
            Title: 'Titel',
            'SHOW MORE': 'ZEIG MEHR',
            'SHOW LESS': 'ZEIGE WENIGER',
            'Do you want to join': 'Wollen Sie der Gruppe betreiten: ',
            Cancel: 'Abbrechen',
            Join: 'Beitreten',
            'Create new post': 'Neuen Beitrag schreiben',
            'Emergency numbers': 'Notrufnummern',
            Comment: 'Kommentar',
            'Create comment': 'Kommentar erstellen',
            'Write comment': 'Kommentar schreiben',
            'Type a message': 'Geben Sie eine Nachricht ein',
            'New message': 'Neue Meldung',
            Search: 'Suche',
            'Save changes': 'Änderungen speichern',
            'Read more': 'Weiterlesen',
            'Read less': 'Lese weniger',
            'Create post': 'Beitrag schreiben',
            'Write some description': 'Schreiben Sie eine Beschreibung',
            'Enter GroupPost title': 'Gruppenpost-Titel eingeben',
            'Name of recipient': 'Name des Empfängers'
        }
    }
};

i18n.use(initReactI18next).init({
    resources,
    lng: 'en', // the default language
    fallbackLng: 'en', //english will be used as language, if other language not found
    interpolation: {
        escapeValue: false
    }
});

export default i18n;
