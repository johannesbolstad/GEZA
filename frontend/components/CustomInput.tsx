import React from 'react';
import { StyleSheet, View } from 'react-native';
import { IconButton, TextInput } from 'react-native-paper';
import { speakUp } from '../translation';

interface InputProps {
    left?: any;
    label: string;
    value: string;
    setValue?: (val: any) => void;
    onPressIn?: (val: any) => void;
    keyboardType: any;
    maxLength: number;
    returnKeyType?: any;
    showSoftInputOnFocus?: boolean;
    language: string;
}

/**
 * This is a custom button that takes ten props
 * @param left a elemetn of type React.ReactNode which is displayed on the left of the TextInput component.
 * @param label the text to use for the floating label.
 * @param value the value of the text input.
 * @param setValue a function that is called when the text inputsÂ´s text changes.
 * @param onPressIn a function that gets called when a touch is engaged.
 * @param keyboardType Decides which keyboard that opens on evnts.
 * @param maxLength max number of characters that can be entered.
 * @param returnKeyType Decides how the return key should look. Options: done/go/next/search/send
 * @param showSoftInputOnFocus if true this will prevent the soft keyboard from showing when the field is focused.
 * @param language the language the SpeakUp function will speak with TTS
 *
 */

const CustomInput: React.FC<InputProps> = (props) => {
    return (
        <View style={styles.subcontainer}>
            <TextInput
                testID="MyInput"
                label={props.label}
                left={props.left}
                onChangeText={props.setValue}
                onPressIn={props.onPressIn}
                mode="outlined"
                style={styles.input}
                value={props.value}
                maxLength={props.maxLength}
                keyboardType={props.keyboardType}
                returnKeyType={props.returnKeyType}
                showSoftInputOnFocus={props.showSoftInputOnFocus}
            />

            <IconButton
                style={styles.icon}
                testID="MyIconButton"
                icon="volume-high"
                size={35}
                onPress={() => speakUp(props.language, props.label)}
            />
        </View>
    );
};

const styles = StyleSheet.create({
    input: {
        height: 60,
        margin: 10,
        marginRight: 1,
        marginLeft: 30,
        borderColor: 'black',
        width: 300
    },
    icon: {
        margin: 0,
        marginTop: 20
    },
    subcontainer: {
        flexDirection: 'row'
    }
});
export default CustomInput;
