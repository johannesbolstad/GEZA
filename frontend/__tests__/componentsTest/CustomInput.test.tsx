import React from 'react';
import renderer, { act, ReactTestRendererJSON } from 'react-test-renderer';
import CustomInput from '../../components/CustomInput';
import { speakUp } from '../../translation';

jest.mock('../../translation');

describe('CustomInput commponent', () => {
    const onPressIn = jest.fn();
    const setValue = jest.fn();
    const tree = renderer
        .create(
            <CustomInput
                label={'home'}
                value={''}
                keyboardType={undefined}
                maxLength={25}
                language={'en'}
            />
        )
        .toJSON() as ReactTestRendererJSON[];
    const tree2 = renderer.create(
        <CustomInput
            label={'home'}
            value={''}
            keyboardType={undefined}
            maxLength={25}
            language={'en'}
            onPressIn={onPressIn}
            setValue={setValue}
        />
    );

    it('Renders correctly', () => {
        expect(tree).toMatchSnapshot();
    });

    it('should fire onPressIn events', () => {
        const inputField = tree2.root.findByProps({ testID: 'MyInput' }).props;
        act(() => inputField.onPressIn());
        expect(onPressIn).toHaveBeenCalled();
    });

    it('Should call setValue when text is change', () => {
        const inputField = tree2.root.findByProps({ testID: 'MyInput' }).props;
        act(() => inputField.onChangeText());
        console.log(inputField.type);
        expect(setValue).toHaveBeenCalled();
    });

    it('icon button should call SpeakUp function when pressed', async () => {
        speakUp as jest.Mock;
        const iconButton = tree2.root.findByProps({
            testID: 'MyIconButton'
        }).props;
        act(() => iconButton.onPress());
        expect(speakUp).toBeCalled();
    });
});
