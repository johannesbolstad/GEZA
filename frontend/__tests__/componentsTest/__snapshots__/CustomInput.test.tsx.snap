// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput commponent Renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "black",
        "margin": 10,
        "marginLeft": 30,
        "marginRight": 1,
        "width": 300,
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          Object {
            "backgroundColor": "#f6f6f6",
            "borderColor": "rgba(0, 0, 0, 0.54)",
            "borderRadius": 4,
            "borderWidth": 1,
          },
        ]
      }
      testID="text-input-outline"
    />
    <View>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 0,
            },
            Object {
              "minHeight": 60,
              "paddingTop": 8,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "zIndex": 3,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "backgroundColor": "#f6f6f6",
                "bottom": 4,
                "left": 10,
                "maxHeight": 2,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": 12,
              }
            }
          />
          <Text
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "#f6f6f6",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 18,
                "maxWidth": -28,
                "opacity": 0,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 39,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "scaleY": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            home
          </Text>
          <Text
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            onLayout={[Function]}
            style={
              Object {
                "color": "#6200ee",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 38,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            home
          </Text>
          <Text
            maxFontSizeMultiplier={1.5}
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 38,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            home
          </Text>
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          maxFontSizeMultiplier={1.5}
          maxLength={25}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=""
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          rejectResponderTermination={true}
          selectionColor="#6200ee"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "zIndex": 1,
              },
              Object {
                "height": 60,
              },
              Object {
                "paddingBottom": 22,
                "paddingTop": 22,
              },
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              Array [
                Object {},
              ],
            ]
          }
          testID="MyInput"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": undefined,
      }
    }
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "margin": 6,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 26.25,
            "height": 52.5,
            "width": 52.5,
          },
          undefined,
          Object {
            "margin": 0,
            "marginTop": 20,
          },
        ],
      ]
    }
    testID="MyIconButton"
  >
    <View>
      <Text />
    </View>
  </View>
</View>
`;
